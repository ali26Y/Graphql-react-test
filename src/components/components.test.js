import React from 'react';
import Footer from './footer';
import Header from './header';
import Tabs from './tabs';
import Renderer from 'react-test-renderer';

const mocks = [
  {
    'title': 'How long can I expect to get my claims resolved?',
    'body': 'A typical claim depends on the nature and time.'
  },
  {
    'title': 'Why is my points not updating?',
    'body': '<p>This because you have money owing or your account is suspended.</p>'
  }
]

describe('Components should match snapshots', () => {
  it('Header matches snapshot', () => {
    const header = Renderer
      .create(<Header />)
      .toJSON();
    expect(header).toMatchSnapshot();
  });

  it('footer matches snapshot', () => {
    const footer = Renderer
      .create(<Footer />)
      .toJSON();
    expect(footer).toMatchSnapshot();
  });
});

describe('Tabs should match mock', () => {
  it('should render mock payload', () => {
    const renderTabs = Renderer
      .create(<Tabs faqs={mocks} />)
      .toJSON();
    expect(renderTabs).toMatchSnapshot();
  })
})

