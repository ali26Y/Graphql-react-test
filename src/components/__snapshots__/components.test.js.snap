// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components should match snapshots Header matches snapshot 1`] = `
<nav
  className="App-header navbar navbar-default"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <img
        alt="qantas-assure-logo"
        className="App-logo navbar-brand"
        src="test-file-stub"
      />
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <ul
        className="nav navbar-nav"
        role={null}
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="/faqs"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Faqs
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Components should match snapshots footer matches snapshot 1`] = `
<div
  className="App-footer"
>
  <p>
    Copyright Â© 2018 Ali. Attayee, All Rights Reserved.
  </p>
</div>
`;

exports[`Tabs should match mock should render mock payload 1`] = `
<div
  className="App-tabs row"
  id="App-tabs"
>
  <div
    className="col-sm-8"
  >
    <div
      className="tab-content"
    >
      <div
        aria-hidden={true}
        aria-labelledby="App-tabs-tab-#0"
        className="fade tab-pane"
        id="App-tabs-pane-#0"
        role="tabpanel"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "A typical claim depends on the nature and time.",
            }
          }
        />
      </div>
      <div
        aria-hidden={false}
        aria-labelledby="App-tabs-tab-#1"
        className="fade tab-pane active in"
        id="App-tabs-pane-#1"
        role="tabpanel"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>This because you have money owing or your account is suspended.</p>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-sm-4"
  >
    <ul
      className="nav nav-pills nav-stacked"
      role="tablist"
    >
      <li
        className=""
        role="presentation"
      >
        <a
          aria-controls="App-tabs-pane-#0"
          aria-selected={false}
          href="#"
          id="App-tabs-tab-#0"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="tab"
          tabIndex={-1}
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "How long can I expect to get my claims resolved?",
              }
            }
          />
        </a>
      </li>
      <li
        className="active"
        role="presentation"
      >
        <a
          aria-controls="App-tabs-pane-#1"
          aria-selected={true}
          href="#"
          id="App-tabs-tab-#1"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="tab"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "Why is my points not updating?",
              }
            }
          />
        </a>
      </li>
    </ul>
  </div>
</div>
`;
